scripts/Adapters/DeployArbAdapter.sol

```diff
@@ -1,42 +1,94 @@
-          abi.encode(
-            arbArgs.baseArgs.crossChainController,
-            arbArgs.inbox,
-            arbArgs.destinationCCC,
-            arbArgs.baseArgs.providerGasLimit,
-            arbArgs.baseArgs.trustedRemotes
-          )
+    bytes memory creationCode = arbArgs.baseArgs.isTestnet
+      ? type(ArbitrumAdapterTestnet).creationCode
+      : type(ArbAdapter).creationCode;
+
+    return
+      abi.encodePacked(
+        creationCode,
+        abi.encode(
+          arbArgs.baseArgs.crossChainController,
+          arbArgs.inbox,
+          arbArgs.destinationCCC,
+          arbArgs.baseArgs.providerGasLimit,
+          arbArgs.baseArgs.trustedRemotes
+        )
+      );
+  }
+}
+
+abstract contract BaseDeployArbAdapter is BaseAdapterScript {
+  function INBOX() internal view virtual returns (address) {
+    return address(0);
+  }
+
+  function _getConstructorArgs(
+    address crossChainController,
+    IBaseAdapter.TrustedRemotesConfig[] memory trustedRemotes
+  ) internal view returns (ArbAdapterDeploymentHelper.ArbAdapterArgs memory) {
+    require(crossChainController != address(0), 'CCC needs to be deployed');
+
+    Addresses memory remoteAddresses;
+    if (isTestnet()) {
+      if (TRANSACTION_NETWORK() == TestNetChainIds.ETHEREUM_SEPOLIA) {
+        remoteAddresses = _getAddresses(TestNetChainIds.ARBITRUM_SEPOLIA);
+        require(
+          remoteAddresses.crossChainController != address(0),
+          'Arbitrum CCC must be deployed'
         );
+        require(INBOX() != address(0), 'Arbitrum inbox can not be 0');
+      }
     } else {
-      return
-        abi.encodePacked(
-          type(ArbAdapter).creationCode,
-          abi.encode(
-            arbArgs.baseArgs.crossChainController,
-            arbArgs.inbox,
-            arbArgs.destinationCCC,
-            arbArgs.baseArgs.providerGasLimit,
-            arbArgs.baseArgs.trustedRemotes
-          )
+      if (TRANSACTION_NETWORK() == ChainIds.ETHEREUM) {
+        remoteAddresses = _getAddresses(ChainIds.ARBITRUM);
+        require(
+          remoteAddresses.crossChainController != address(0),
+          'Arbitrum CCC must be deployed'
         );
+        require(INBOX() != address(0), 'Arbitrum inbox can not be 0');
+      }
     }
+
+    return
+      ArbAdapterDeploymentHelper.ArbAdapterArgs({
+        baseArgs: BaseAdapterArgs({
+          crossChainController: crossChainController,
+          providerGasLimit: PROVIDER_GAS_LIMIT(),
+          trustedRemotes: trustedRemotes,
+          isTestnet: isTestnet()
+        }),
+        inbox: INBOX(),
+        destinationCCC: remoteAddresses.crossChainController
+      });
+  }
+
+  function _deployAdapter(
+    Addresses memory addresses,
+    IBaseAdapter.TrustedRemotesConfig[] memory trustedRemotes
+  ) internal override returns (address) {
+    ArbAdapterDeploymentHelper.ArbAdapterArgs memory constructorArgs = _getConstructorArgs(
+      addresses.crossChainController,
```

Comment:
At least looks like mix, because here addresses enters from outside, but inside you use transaction_network